std/datetime/stopwatch.d

```diff
@@ -349,6 +349,13 @@ private:
     long _ticksElapsed;    // Total time that the StopWatch ran before it was stopped last.
 }
 
+private void doNotOptimizeAway(T)(auto ref T t) @trusted @nogc nothrow
+{
+    import core.thread : getpid;
+    import core.stdc.stdio;
+    if(getpid() == 1)
```

Comment:
I have the following idea for the unoptimizable test:

```d
public __gshared void* _13f43be984760828c93138284d104611;
private void doNotOptimizeAway(T)(auto ref T t) @trusted @nogc nothrow
{
    if (&t == _13f43be984760828c93138284d104611) { ... }
}
```

Although undocumented and obfuscated, that global is technically public so the compiler must conservatively assume it may point anywhere within the address space, including the address of the parameter.

Works? cc @WalterBright @ibuclaw @klickverbot